index.js

```diff
@@ -27,29 +30,96 @@ const urls = fs
-  'extensions.webextensions.uuids',
-  `{"firefox@ghostery.com": "${addonUUID}"}`,
-);
-// options.addArguments("--headless");
+if (isChromeSelected) {
+  addonUUID = 'agbomdmjjfglgplpnkahbcmblehajkag';
+}
+
+let options = new firefox.Options();
+let selectedBrowser = 'Firefox';
 
-if (isGhosteryEnabled) {
-  options.addArguments('-profile', 'profiles/withGhostery');
-  outputPath += '/withGhostery';
+if (isChromeSelected) {
+  selectedBrowser = 'Chrome';
+  console.log(`LOG: Selected browser: ${selectedBrowser}.`);
+  options = new chrome.Options();
 } else {
-  options.addArguments('-profile', 'profiles/withoutGhostery');
-  outputPath += '/withoutGhostery';
+  console.log(`LOG: Selected browser: ${selectedBrowser}.`);
+  options.setPreference(
+    'extensions.webextensions.uuids',
+    `{"firefox@ghostery.com": "${addonUUID}"}`,
+  );
+  // options.addArguments("--headless");
 }
 
-const outputStream = fs.createWriteStream(`${outputPath}/${timestamp}.txt`);
+let outputPath = 'output/time';
+let driver = '';
+let addon = '';
 
-const driver = await new Builder()
-  .forBrowser(Browser.FIREFOX)
-  .setFirefoxOptions(options)
-  .build();
+if (isChromeSelected) {
+  if (isGhosteryEnabled) {
+    options.addArguments('--profile-directory=Default');
+    options.addArguments(
+      `--user-data-dir=profiles/withGhostery/${selectedBrowser}`,
+    );
+    outputPath += `/withGhostery/${selectedBrowser}`;
+
+    console.log(`LOG: Installing addon for ${selectedBrowser}.`);
+    addon = await downloadAddon(
+      'https://github.com/ghostery/ghostery-extension/releases/download/v10.2.10/ghostery-chrome.zip',
+    );
+
+    options.addArguments(`--load-extension=${addon}`);
+    driver = await new Builder()
+      .forBrowser(Browser.CHROME)
+      .setChromeOptions(options)
+      .build();
+  } else {
+    options.addArguments('--profile-directory=Default');
+    options.addArguments(
+      `--user-data-dir=profiles/withoutGhostery/${selectedBrowser}`,
+    );
+    outputPath += `/withoutGhostery/${selectedBrowser}`;
+
+    driver = await new Builder()
+      .forBrowser(Browser.CHROME)
+      .setChromeOptions(options)
+      .build();
+  }
+} else {
+  if (isGhosteryEnabled) {
+    options.addArguments(
+      '-profile',
+      `profiles/withGhostery/${selectedBrowser}`,
+    );
+    outputPath += `/withGhostery/${selectedBrowser}`;
+
+    driver = await new Builder()
+      .forBrowser(Browser.FIREFOX)
+      .setFirefoxOptions(options)
+      .build();
+
+    console.log(`LOG: Installing addon for ${selectedBrowser}.`);
+    addon = await downloadAddon(
+      'https://github.com/ghostery/ghostery-extension/releases/download/v10.2.10/ghostery-firefox.zip',
+    );
+
+    await driver.installAddon(addon, true);
+  } else {
+    options.addArguments(
+      '-profile',
+      `profiles/withoutGhostery/${selectedBrowser}`,
+    );
+    outputPath += `/withoutGhostery/${selectedBrowser}`;
+
+    driver = await new Builder()
```

Comment:
this is pretty much repeated 4 time, can be refactored so the Builder is instantiated once with different params.

eg:
```js
const options = ? new firefox.Options() : new chrome.Options();
const driverBuilder = new Builder();

if (isFirefox) {
  const option = new firefox.Options();
  options.addAgrguments(...)
  driverBuilder.forBrowser(Browser.FIREFOX)
  driverBuilder.setFirefoxOptions(options)
} else {
  ... chrome stuff
}
const driver = await driver.builder();
```